src/appleseed/renderer/kernel/lighting/imagebasedlighting.cpp

```diff
@@ -207,16 +207,23 @@ void compute_ibl_environment_sampling(
 
         // Discard occluded samples.
         Spectrum transmission;
-        const ShadingPoint& shading_point = material_sampler.trace(
-            shading_context,
-            incoming,
-            transmission);
+        if(environment_edf.get_cast_shadows())
+        {
+            const ShadingPoint& shading_point = material_sampler.trace(
+                shading_context,
+                incoming,
+                transmission);
 
-        if (shading_point.hit_surface())
-            continue;
-        
-        if (max_value(transmission) == 0.0f)
-            continue;
+            if (shading_point.hit_surface()){}
```

Comment:
You have a very serious bug here (the `{}` at the end).

Did you test your changes?